File: GameLibrary/src/cc/lib/reflector/Archiver.java
package cc.lib.reflector;

import java.io.IOException;
import java.lang.reflect.Field;

interface Archiver {
    String get(Field field, Reflector<?> a) throws Exception;

    void set(Object o, Field field, String value, Reflector<?> a, boolean keepInstances) throws Exception;

    void serializeArray(Object arr, RPrintWriter out) throws IOException;

    void deserializeArray(Object arr, RBufferedReader in, boolean keepInstances) throws IOException;
}

File: GameLibrary/src/cc/lib/kreflector/Archiver.kt
package cc.lib.kreflector

import java.io.IOException
import java.lang.reflect.Field

interface Archiver {
	@Throws(Exception::class)
	operator fun get(field: Field, a: Reflector<*>): String

	@Throws(Exception::class)
	operator fun set(o: Any?, field: Field, value: String, a: Reflector<*>, keepInstances: Boolean)

	@Throws(IOException::class)
	fun serializeArray(arr: Any, out: RPrintWriter)

	@Throws(IOException::class)
	fun deserializeArray(arr: Any, reader: RBufferedReader, keepInstances: Boolean)
}